solaris/src/evm.rs

```diff
@@ -0,0 +1,213 @@
+            value: 0.into(),
+            logs: vec![],
+        }
+    }
+
+    fn tracers(&self) -> (trace::PrintingTracer, trace::PrintingTracer) {
+        Default::default()
+    }
+
+    fn env_info(&self) -> client::EnvInfo {
+        client::EnvInfo {
+            number: 1u64,
+            author: 0.into(),
+            timestamp: 1u64,
+            difficulty: 1.into(),
+            last_hashes: Arc::new([0.into(); 256].to_vec()),
+            gas_used: 0.into(),
+            gas_limit: 4_700_000.into(),
+        }
+    }
+
+    pub fn deploy(&mut self, code: &[u8]) -> Result<Address, ()> {
+        let env_info = self.env_info();
+        let nonce = self.evm.state().nonce(&convert_address(self.sender)).expect(STATE);
+        let transaction = transaction::Transaction {
+            nonce,
+            gas_price: convert_u256(self.gas_price),
+            gas: convert_u256(self.gas),
+            action: transaction::Action::Create,
+            value: convert_u256(self.value),
+            data: code.to_vec(),
+        }.fake_sign((&*self.sender).into());
+
+        let tracers = self.tracers();
+        match self.evm.transact(&env_info, transaction, tracers.0, tracers.1) {
+            TransactResult::Ok { contract_address, .. } => {
+                self.contract_address = contract_address.map(|x| (&*x).into());
+                Ok(Default::default())
+            },
+            TransactResult::Err { .. } => {
+                Err(())
+            },
+        }
+    }
+
+    pub fn with_gas(&mut self, gas: U256) -> &mut Self {
+        self.gas = gas;
+        self
+    }
+
+    pub fn with_gas_price(&mut self, gas_price: U256) -> &mut Self {
+        self.gas_price = gas_price;
+        self
+    }
+
+    pub fn with_value(&mut self, value: U256) -> &mut Self {
+        self.value = value;
+        self
+    }
+
+    pub fn with_sender(&mut self, address: Address) -> &mut Self {
+        self.sender = address;
+        self
+    }
+
+    /// Ensures that sender has enough funds (value) to call next transaction.
+    pub fn ensure_funds(&mut self) -> &mut Self {
+        // TODO [ToDr] Just transfer to amount that is actually needed
+        let env_info = self.env_info();
+        let sender = "7c532DB9E0c06C26fd40Acc56AC55C1eE92D3C3A".parse().unwrap();
+        let nonce = self.evm.state().nonce(&sender).expect(STATE);
+        let transaction = transaction::Transaction {
+            nonce,
+            gas_price: 0.into(),
+            gas: 21_000.into(),
+            action: transaction::Action::Call(convert_address(self.sender)),
+            value: convert_u256(self.value),
+            data: vec![],
+        }.fake_sign(sender);
+
+        self.evm.transact(&env_info, transaction, ethcore::trace::NoopTracer, ethcore::trace::NoopVMTracer);
+
+        self
+    }
+
+    pub fn logs(&self, _filter: ::ethabi::TopicFilter) -> Vec<()> {
+        // TODO [ToDr] Add filter querying
+        self.logs.iter().map(|_| ()).collect()
```

Comment:
Is this just acting as a stub for now?

Reply:
Yes, should do actual filtering by topics from `ethabi::TopicFilter`.